### PR TITLE
Add OpenEBS k8s provisoner

### PR DESCRIPTION
**Pull request by satyamz**
This will add OpenEBS kubernetes provisioner to kubernetes incubator/external-storage project.

**Tasks covered:** 

- Add OpenEBS Provisioner.
- Add OpenEBS provisioner related make targets to external-storage/Makefile.  
- Allow the user to build Docker image.
- Allow the user to push Docker image to Docker hub by accepting user details, image name etc. 
- Add initial documentation related to building and shipping openebs-provisioner. 

**Tasks Remaining:** 
- Add tests for openebs/pkg/maya_api.go and openebs/openebs-provisioner.go

### CODE REVIEW

File: openebs/.travis.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+script:
+  - make bootstrap
+  - make deps
+  - make build
+after_success:
+  - make image
+notifications:
+  email:
+    recipients:
+    - kiran.mova@cloudbyte.com
```
Comment (kmova) [inline openebs/.travis.yml]: change the email id to openebs.io

File: openebs/openebs-provisioner.go [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+	openEBSProvisioner := NewOpenEBSProvisioner(clientset)
+
+	// Start the provision controller which will dynamically provision OpenEBS VSM
+	// PVs
+	pc := controller.NewProvisionController(
+		clientset,
+		provisionerName,
+		openEBSProvisioner,
+		serverVersion.GitVersion)
+	// resyncPeriod,
```
Comment (kmova) [inline openebs/openebs-provisioner.go]: remove the commented code.

File: openebs/pkg/v1/maya_api.go [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+	if err != nil {
+		glog.Fatalf("http.Do() error: : %v", err)
+		return err
+	}
+	defer resp.Body.Close()
+
+	code := resp.StatusCode
+	if code != http.StatusOK {
+		glog.Fatalf("Status error: %v\n", http.StatusText(code))
+		os.Exit(1)
```
Comment (kmova) [inline openebs/pkg/v1/maya_api.go]: return an error here instead of exiting

File: openebs/pkg/v1/maya_api.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
```
Comment (kmova) [inline openebs/pkg/v1/maya_api.go]: return error instead of exiting

File: openebs/openebs-provisioner.go [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+			iqn = value.(string)
+		case "vsm.openebs.io/targetportals":
+			targetPortal = value.(string)
+		}
+	}
+
+	glog.Infof("Volume IQN: %v", iqn)
+	glog.Infof("Volume Target: %v", targetPortal)
+
+	//TODO - fill in the iSCSI PV details
```
Comment (kmova) [inline openebs/openebs-provisioner.go]: "TODO" is not required before the comment.

File: openebs/.travis.yml (hunk 1 above)
Comment (satyamz) [reply openebs/.travis.yml]: Actually, As of now we don't need `.travis.yaml` file anymore! I am removing it as it is redundant now.

File: openebs/openebs-provisioner.go (hunk 5 above)
Comment (satyamz) [reply openebs/openebs-provisioner.go]: Ok! I will remove this.

File: openebs/openebs-provisioner.go (hunk 2 above)
Comment (satyamz) [reply openebs/openebs-provisioner.go]: Sure! I wanted to know whether we need those extra parameters or not.

File: openebs/pkg/v1/maya_api.go (hunk 3 above)
Comment (satyamz) [reply openebs/pkg/v1/maya_api.go]: Sure!

**Comment by satyamz** [comment]
@kmova  Done! I removed redundant code comments.

File: openebs/Makefile [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+.DEFAULT_GOAL := build
+
+
+
+build: $(shell find . -name "*.go")
+	CGO_ENABLED=0 go build -a -ldflags '-extldflags "-static"' -o openebs-provisioner .
+
+push: 
+	@cp openebs-provisioner buildscripts/docker/
+	@cd buildscripts/docker && sudo docker build -t openebs/openebs-k8s-provisioner:ci .
```
Comment (kmova) [inline openebs/Makefile]: may be we can re-name the docker image name to -- openebs-provisioner (remove k8s)

**Review by kmova — COMMENTED** [review]
Also, should we have a sample k8s yaml file that will deploy the openebs-provisioner on k8s cluster?

**Pull request merged**